src/components/Modal/index.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from "react";
+
+import useDateToken from "hooks/useDateToken";
+
+import { ModalElemProps } from "./ModalElem";
+
+import modalsAtom from "atoms/modals";
+import { useSetRecoilState } from "recoil";
+
+const Modal = (props: ModalElemProps) => {
+  const [id] = useDateToken();
+  const setModals = useSetRecoilState(modalsAtom);
+
+  useEffect(() => {
+    setModals((prev) => [...prev, { id, props }]);
+    return () => setModals((prev) => prev.filter((modal) => modal.id !== id));
+  }, []);
+
+  useEffect(() => {
+    setModals((prev) =>
+      prev.map((modal) => (modal.id === id ? { id, props } : modal))
+    );
+  }, [props]);
+
+  return null;
+};
```

Comment:
오호 이렇게 하는 방식 좋은거 같아요! 근데 props가 바뀔 일이 있을까요?

Reply:
아 isOpen이 바뀌는 군요 그러면 isOpen 만 dep에 넣는 건 어떤가요?

Reply:
아 props가 object 네요. 생각 못했습니다. children 도 바뀔 것 같은데 `JSON.stringify(props)`는 어떨까요??

Reply:
뭔가 매번 stringify를 하는게 이상적이지는 않은거 같은데 조금 더 좋은 구조가 없을까라는 생각이 드네요 흠..

Reply:
다른 모달 리랜더링은

https://github.com/sparcs-kaist/taxi-front/blob/f4412d88d459f57df00ffff7ab22dd39a8c00ceb/src/components/Modal/ModalProvider.tsx#L6-L9

에서 막힐 줄 알았는데, 안막히나요???

Reply:
아 그렇겠네요..? 좋습니다~

Reply:
@14KGun 앜 다시 생각해보니 아닙니다!ㅋㅋㅋㅋ `modals`라는 recoil state가 바뀌기 때문에 `ModalProvider`라는 부모 컴포넌트는 무조건 리랜더가 되네요. 그래서 자식 컴포넌트인 `ModalElem`들도 모두 다 리랜더가 될듯합니다. 그래서 예전에 업급했던 `React.memo`로 한번 감싸주면 변화하지 않는 컴포넌트는 리랜더가 안될것 같습니다ㅎ

Reply:
@imYourChoi 흐으으음,,;; 그렇군요 ㅜㅜ 😲 😲 

그럼 `taxi-front`에서 쓰이는 왠만한 컴포넌트들은 `React.memo` 로 감싸는게 이득일까요??

저걸로 모든 컴포넌트들을 감싸면, axios쓰는 컴포넌트들이 업데이트 안되는 버그가 발생될 것 같기두 하고,
생각나는거는 `useEffect`, `useState`, `useCallback`, `useMemo` 잘 쓰는 거 밖에는 떠오르지가 않네요 ㅜㅜㅜ

Reply:
음.. 저도 사실 잘 모르겠네요..
뭔가 코드의 완성도 측면에서 생각하니까 계속 최적화할 부분이 보여서 얘기하긴 하지만 얼른 구현하는게 중요한 저희로서는 어느 지점에서 타협해야 될지 모르겠네요.

그냥 React.memo 쓴다고 항상 좋은 것도 아니고 간단한 리랜더링이면 닭 잡는데 소 잡는 칼 쓰는 느낌이기도 하구요. 사실 이걸 안쓰는 상황을 만드는게 제일 좋은 것 같아요. 물론 그게 제일 어렵죠.. (위의 경우에는 그냥 ModalElem 정도만 감싸면 괜찮을지도..?)

(그리고 axios 쓰는 컴포넌트 업데이트 안되는건 무슨 이야기인지 이해를 못했는데 설명해주실 수 있나요?)

Reply:
아 네네 저도 그렇게 이해하고 있습니다. 부모의 랜더링에 의한 리랜더는 막아주고 컴포넌트 안에서의 state나 context 변화는 영향 없습니다. 정말 어렵고 이렇게까지 해야되나 싶긴 한데 다 피가 되고 살이 되겠죠뭐..

> Even when a component is memoized, it will still re-render when its own state changes. Memoization only has to do with props that are passed to the component from its parent.